### PR TITLE
Build pattern with Osti for pub link in waterfall

### PR DESCRIPTION
**Pull request by hesspnnl**
Add checking for publication links to the main context of the waterfall

### CODE REVIEW

File: src/nmdc_metadata_suggestor/doi_ingestion/osti.py [hunk 1]
```diff
@@ -42,17 +37,40 @@ def try_osti(doi: str, errors: list[str] | None = None) -> ResolverContext | Non
...
 
     record = data[0] if isinstance(data, list) else data
-    abstract = record.get("description")
-    if not isinstance(abstract, str):
-        append_error(errors, "No abstract/description found in OSTI record")
-        return None
 
-    cleaned = clean_text(abstract)
-    if not cleaned:
+    raw_description = record.get("description")
```
Comment (caufieldjh) [inline src/nmdc_metadata_suggestor/doi_ingestion/osti.py]: It's possible that the record will be retrievable and valid but won't contain much useful text (e.g., https://www.osti.gov/biblio/2423571). In an edge case it may not even be present, and that gets caught in lines 45-46, but it will still be returned as a `ResolverContext` if it gets a PDF link. That might be OK if other sources also don't provide additional text.

**Review by caufieldjh — APPROVED** [review]
I think this is ready to go, but one of the next steps will need to be a broader evaluation than the current integration tests are doing. I'm thinking about at least a few hundred DOIs (maybe from those already used in NMDC entries) to catch edge cases and publication sets unexpectedly missing from one source or another.
